tutoroars/templates/oars/apps/vector/partials/common-post.toml

```diff
@@ -0,0 +1,114 @@
+### Transforms
+
+# Parse tracking logs: extract time
+[transforms.tracking]
+type = "remap"
+inputs = ["openedx_containers"]
+# Time formats: https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html#specifiers
+source = '''
+parsed, err_regex = parse_regex(.message, r'^.* \[tracking\] [^{}]* (?P<tracking_message>\{.*\})$')
+if err_regex != null {
+  abort
+}
+message = parsed.tracking_message
+parsed_json, err_json = parse_json(parsed.tracking_message)
+if err_json != null {
+  log("Unable to parse JSON from tracking log message: " + err_json, level: "error")
+  abort
+}
+time, err_timestamp = parse_timestamp(parsed_json.time, "%+")
+if err_timestamp != null {
+  log("Unable to parse timestamp from tracking log 'time' field: " + err_timestamp, level: "warn")
+  time, err_timestamp = parse_timestamp(parsed_json.timestamp, "%+")
+  if err_timestamp != null {
+    log("Unable to parse timestamp from tracking log 'timestamp' field: " + err_timestamp, level: "error")
+    abort
+  }
+}
+. = {"time": time, "message": message}
+'''
+drop_on_error = true
+drop_on_abort = true
+
+[transforms.tracking_debug]
+type = "remap"
+inputs = ["tracking"]
+# Time formats: https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html#specifiers
+source = '''
+.message = parse_json!(.message)
+'''
+
+[transforms.xapi]
+type = "remap"
+inputs = ["lms_worker"]
+# Time formats: https://docs.rs/chrono/0.4.19/chrono/format/strftime/index.html#specifiers
+
+source = '''
+parsed, err_regex = parse_regex(.message, r'^.* \[xapi_tracking\] [^{}]* (?P<tracking_message>\{.*$)')
+if err_regex != null {
+  abort
+}
+message, err = strip_whitespace(parsed.tracking_message)
+
+parsed_json, err_json = parse_json(parsed.tracking_message)
+
+if err_json != null {
+  log("Unable to parse JSON from xapi tracking log message: " + err_json, level: "error")
+  abort
+}
+
+time, err_timestamp = parse_timestamp(parsed_json.timestamp, "%+")
+if err_timestamp != null {
+  log("Unable to parse timestamp from tracking log 'time' field: " + err_timestamp, level: "warn")
+  time, err_timestamp = parse_timestamp(parsed_json.timestamp, "%+")
+  if err_timestamp != null {
+    log("Unable to parse timestamp from tracking log 'timestamp' field: " + err_timestamp, level: "error")
+    abort
+  }
+}
+event_id = parsed_json.id
+. = {"event_id": event_id, "emission_time": format_timestamp!(time, format: "%+"), "event_str": encode_json(parsed_json)}
```

Comment:
At the moment this table structure differs from the Ralph table I actually want to write to. I was having trouble writing to the JSON column, so omitted it for this initial testing. This needs to be fixed before merging.

Reply:
Or maybe it doesn't, if we don't use the json column for anything else. Ralph uses it, but maybe we don't care?

Reply:
We can make the JSON column with a default value of "{}" and based all the work on the other available field